### PR TITLE
Null-merge branch 5.6 (commit f5c4f8d) into 5.7

### PR DESCRIPTION
**Pull request by inikep**
This is a null merge after merging upstream 5.6.40 and 5.7.22.

### CODE REVIEW

File: sql/sp_head.cc [hunk 1]
```diff
@@ -1973,9 +1973,10 @@ bool sp_head::show_routine_code(THD *thd)
     */
     if (ip != i->get_ip())
     {
-      const char *format= "Instruction at position %u has m_ip=%u";
-      char tmp[64 + 2 * MY_INT32_NUM_DECIMAL_DIGITS];
-      sprintf(tmp, format, ip, i->get_ip());
+      const char format[]= "Instruction at position %u has m_ip=%u";
```
Comment (laurynas-biveinis) [inline sql/sp_head.cc]: If this is touched, might as well be made static

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
It is slightly suspicious that the warning fixes, which motivated the changes being ported, need re-doing in 5.7?

**Comment by inikep** [comment]
5.5 and 5.6 had a valid warning because it was defined: 
```
const char *format = ...
char tmp[sizeof(format) + 2 * sizeof(uint) + 1];
```

5.7 had no warning because of:
```
const char *format = ...
char tmp[64 + 2 * MY_INT32_NUM_DECIMAL_DIGITS];
```

**Comment by laurynas-biveinis** [comment]
So then it should be a null merge - cleanup patches without a clear objective (e.g. a warning fix) is not something we should maintain

**Comment by inikep** [comment]
Changes within `sp_head.cc` were removed.

**Comment by laurynas-biveinis** [comment]
The question still stands for the remaining changes

**Comment by dutow** [comment]
As I see, the warning fixes are mostly about the register keyword.

* In 5.6, I included the commit which removed them (180b7e4). That commit is also part of the current 5.7 branch.
* However, in 5.7 commits after 180b7e4 again added the register keyword to some locations - so its not a merge issue, they are simply newer than the commit which removes them.

This shouldn't result in any warnings/errors in 5.7 (that's why I didn't notice them when I fixed the 5.7 warnings), as we always build 5.7 in C++03/11 mode (compared to 5.5/5.6, where we use the default language level of the compiler) to avoid some other deprecation warnings in the standard library.

**Comment by inikep** [comment]
After removing changes without any visible payoff the null-merge is empty.

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**